packages/devtools_app/test/test_infra/scenes/standalone_ui/editor_sidebar.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/standalone_ui/vs_code/flutter_panel.dart';
+import 'package:devtools_app_shared/service.dart';
+import 'package:devtools_app_shared/shared.dart';
+import 'package:devtools_app_shared/ui.dart';
+import 'package:devtools_app_shared/utils.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:dtd/dtd.dart';
+import 'package:flutter/material.dart';
+import 'package:stager/stager.dart';
+
+import 'editor_service/fake_editor.dart';
+import 'mock_editor_widget.dart';
+import 'utils.dart';
+
+/// To run, use the "standalone_ui/editor_sidebar" launch configuration with the
+/// `devtools/packages/` folder open in VS Code, or run:
+///
+///   flutter run -t test/test_infra/scenes/standalone_ui/editor_sidebar.stager_app.g.dart --dart-define=enable_experiments=true -d chrome
+class EditorSidebarScene extends Scene {
+  late PerformanceController controller;
+  late Stream<String> clientLog;
+  late DartToolingDaemon clientDtd;
+  late FakeDtdEditor editor;
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: themeFor(
+        isDarkTheme: false,
+        ideTheme: _ideTheme(const VsCodeTheme.light()),
+        theme: ThemeData(useMaterial3: true, colorScheme: lightColorScheme),
+      ),
+      darkTheme: themeFor(
+        isDarkTheme: true,
+        ideTheme: _ideTheme(const VsCodeTheme.dark()),
+        theme: ThemeData(useMaterial3: true, colorScheme: darkColorScheme),
+      ),
+      home: Scaffold(
+        body: MockEditorWidget(
+          editor: editor,
+          clientLog: clientLog,
+          child: DtdEditorSidebarPanel(clientDtd),
+        ),
+      ),
+    );
+  }
+
+  /// Creates an [IdeTheme] using the colours from the mock editor.
+  IdeTheme _ideTheme(VsCodeTheme vsCodeTheme) {
+    return IdeTheme(
+      backgroundColor: vsCodeTheme.editorBackgroundColor,
+      foregroundColor: vsCodeTheme.foregroundColor,
+      embedMode: EmbedMode.embedOne,
+    );
+  }
+
+  @override
+  String get title => '$EditorSidebarScene';
+
+  @override
+  Future<void> setUp() async {
+    setStagerMode();
+    setGlobal(
+      DevToolsEnvironmentParameters,
+      ExternalDevToolsEnvironmentParameters(),
+    );
+    setGlobal(DTDManager, MockDTDManager());
+    setGlobal(IdeTheme, IdeTheme());
+    setGlobal(PreferencesController, PreferencesController());
+
+    // We assume a DTD is available on 8500. There's a VS Code task that
+    // launches this as part of the standalone_ui/editor_sidebar config.
+    final dtdUri = Uri.parse('ws://127.0.0.1:8500/');
```

Comment:
It does not currently - but it's probably something we'll want to support. I'll add a TODO because I'm not sure exactly what we want here (it would be nice if you could set the workspace roots from buttons in the simulated editor) and this PR is already quite large.

(TODO added in 60707d74c)